Docs: Custom PK in fulltext index (#43368)
ydb-platform/ydb: ydb/docs/en/core/dev/fulltext-indexes.md

A `LIKE` / `ILIKE` query uses the same logic as `FulltextMatch(body, ..., "Wildcard" AS Mode)` and accesses the same n-gram index.

## Primary key types {#primary-key}

Inside the inverted index, every indexed document is identified by a numeric document id (`doc_id`). How {{ ydb-short-name }} derives the `doc_id` depends on the base table's [primary key](../concepts/glossary.md#primary-key):

* **Single integer primary key** (`Uint64`, `Int64`, `Uint32`, or `Int32`) — the primary key value is used directly as the `doc_id`. This is the most compact form, and no additional structures are created.

> * **Any other primary key** (for example `Utf8` / `String`, a non-integer type, or a composite key of several columns) — {{ ydb-short-name }} maintains a system column `__ydb_row_id` of type `Uint64` and uses it as the `doc_id`.

Reviewer: ```suggestion
* **Any other primary key** (for example, `Utf8`, `String`, other non-integer types, or a composite key of several columns) — {{ ydb-short-name }} maintains a system column `__ydb_row_id` of type `Uint64` and uses it as the `doc_id`.
```

When you create a fulltext index on a table whose primary key is not a single integer column, {{ ydb-short-name }} automatically:

* adds the `__ydb_row_id` column to the table — existing rows are backfilled while the index is being built;

> * generates `__ydb_row_id` values automatically as rows are inserted (you never set this column yourself);

Author: \+

> * generates a `__ydb_row_id` value automatically for every row where the column is omitted from `INSERT` / `UPSERT` (you normally don't set it yourself);

Reviewer: ```suggestion
* generates a `__ydb_row_id` value for every row where the column is omitted from `INSERT` / `UPSERT`;
```

## Limitations {#limitations}

> * Any primary key type is supported. Tables whose primary key is not a single integer column get an auto-managed `__ydb_row_id` system column and a `uniq__ydb_row_id` unique index (see [Primary key types](#primary-key)).

Author: \+

> * Any primary key type is supported. For a primary key that is not a single integer column, {{ ydb-short-name }} auto-provisions an `__ydb_row_id` system column and a `uniq__ydb_row_id` unique index (see [Primary key types](#primary-key)).

Reviewer: ```suggestion
* Tables with a non-integer or composite primary key get an auto-managed `__ydb_row_id` column and `uniq__ydb_row_id` unique index (see [Primary key types](#primary-key)).
```
